PICMI_Python/base.py

```diff
@@ -51,3 +53,37 @@ def init(self, kw):
         # --- The implementation of this routine should use kw.pop() to retrieve input arguments from kw.
         # --- This allows testing for any unused arguments and raising an error if found.
         pass
+
+    def _check_unsupported_argument(self, arg_name, extra_info=None, raise_error=False):
+        """Raise a warning or exception for an unsupported argument."""
+        # --- Implementation note: This should be called in the "init" method of the implementing
+        # --- class for each unsupported argument. For example, for the 'density_scale' argument of Species:
+        # ---     self._check_unsupported_argument('density_scale', extra_info='My code can not handle a density_scale')
+
+        # This compares the value of the parameter with the dault value in the __init__ method.
+        # If they differ, this means that the user supplied a value, so a warning or error is raised.
+        signature = inspect.signature(self.__init__)
+        default_value = signature.parameters[arg_name].default
+        if not (getattr(self, arg_name) == default_value):
+            message = f'{self.__name__}: For argument {arg_name} is not supported.'
+            if extra_info is not None:
+                message += f' {extra_info}'
+            if raise_error:
+                raise Exception(message)
+            else:
+                warnings.warn(message)
+
+    def _unsupported_value(self, arg_name, extra_info='', raise_error=True):
+        """Raise a warning or exception for argument with an unsupported value."""
```

Comment:
This needs parameter documentations please :)

Reply:
@ax3l Done